### PR TITLE
feat: collect metrics from tunnel api via opentelemetry

### PR DESCRIPTION
**Pull request by ternakkode**
This pull request introduces a new feature to enhance our application's observability by integrating metrics collection using the OpenTelemetry SDK. With this addition, we can gather valuable insights into the performance and behavior of our application.

Key Changes:

Implemented metrics collection using the OpenTelemetry SDK.
Configured Telegraf to listen for metrics on port 4317 and forward them to InfluxDB.
Updated relevant documentation to reflect the changes and provide instructions for setup.

### CODE REVIEW

**Review by DeeStarks — APPROVED** [review]
Nice work! I haven't tested yet, I'll set it up and test but looks good

**Review by NJavokhir — COMMENTED** [review]
I'm having trouble generating an access token in the InfluxDB UI on my own side. Overall, good work!

File: README.md [hunk 1]
```diff
@@ -118,3 +118,26 @@ app.post("/route", async (req, res) => {
...
+```
+2. Open the InfluxDB dashboard via a browser using the defined credentials in docker-compose-influx.yml on port 8086.
+3. Create an access token in the InfluxDB UI (https://docs.influxdata.com/influxdb/cloud/admin/tokens/create-token/).
+4. Edit the `INFLUXDB_URL` and `INFLUXDB_TOKEN` values in the docker-compose-telegraf.yml file based on the created token.
+5. Start Telegraf by running the following command:
+```
+docker compose -f docker-compose-telegraf.yml up 
+```
+6. After Telegraf is up and running, any metrics collected by the OpenTelemetry SDK could be sent via the gRPC protocol to port 4317.
+7. For our case, set `OTEL_EXPORTER_OTLP_ENDPOINT` to localhost:4317.
```
Comment (stravid87) [inline README.md:143]: Is this variable, `OTEL_EXPORTER_OTLP_ENDPOINT` to be added to the ./server/.env.dev ? file?

Comment (huzaifamk) [reply README.md]: Had same question here.

File: README.md [hunk 2]
```diff
@@ -118,3 +118,27 @@ app.post("/route", async (req, res) => {
...
+### Setup
+
+1. Start InfluxDB v2 as a Docker container by running the following command:
+```
+docker compose -f docker-compose-influx.yml up 
+```
+2. Open the InfluxDB dashboard via a browser using the defined credentials in docker-compose-influx.yml on port 8086.
+3. Create an access token in the InfluxDB UI (https://docs.influxdata.com/influxdb/cloud/admin/tokens/create-token/).
+4. Create a bucket with `layer8` name in InfluxDB
+5. Add `INFLUXDB_URL` and `INFLUXDB_TOKEN` variable to `.env` file based on the created token.
```
Comment (huzaifamk) [inline README.md:138]: Please update the .env.dev file accordingly, so it is easier for the team members to know which environment variables should be made.

File: README.md [hunk 3]
```diff
@@ -118,3 +118,27 @@ app.post("/route", async (req, res) => {
...
+6. Start Telegraf by running the following command:
+```
+docker compose -f docker-compose-telegraf.yml up 
+```
+7. After Telegraf is up and running, any metrics collected by the OpenTelemetry SDK could be sent via the gRPC protocol to port 4317.
+8. For our case, set `OTEL_EXPORTER_OTLP_ENDPOINT` to localhost:4317.
```
Comment (huzaifamk) [inline README.md:144]: Same goes for here, please update the .env.dev

File: README.md (hunk 2 above)
Comment (huzaifamk) [inline README.md:138]: Please update the .env.dev file accordingly, so it is easier for the team members to know which environment variables should be made.

**Review by huzaifamk — CHANGES_REQUESTED** [review]

**Comment by huzaifamk** [comment]
@ternakkode I tried everything according to the README, and setup was done, but during testing, I was getting this error in the telegraf:

image

And following error in the influx:
image

I already made the .env in the root also, and in the server, both of them, but getting these error.

File: README.md (hunk 1 above)
Comment (ternakkode) [reply README.md]: yes we should add it to layer8 server env variable

**Comment by ternakkode** [comment]
> @ternakkode I tried everything according to the README, and setup was done, but during testing, I was getting this error in the telegraf:
> 
> image
> 
> And following error in the influx: image
> 
> I already made the .env in the root also, and in the server, both of them, but getting these error.



> @ternakkode I tried everything according to the README, and setup was done, but during testing, I was getting this error in the telegraf:
> 
> image
> 
> And following error in the influx: image
> 
> I already made the .env in the root also, and in the server, both of them, but getting these error.



> @ternakkode I tried everything according to the README, and setup was done, but during testing, I was getting this error in the telegraf:
> 
> image
> 
> And following error in the influx: image
> 
> I already made the .env in the root also, and in the server, both of them, but getting these error.

Hi @huzaifamk, you shoud put `host.docker.internal` as the INFLUXDB_URL instead of `localhost`

**Comment by huzaifamk** [comment]
@ternakkode I can get this merged if you update the .env.dev and also the README file steps, the `host.docker.internal` and other stuff. I will then test again and will merge if everything is OK.

**Review by huzaifamk — APPROVED** [review]
Good work @ternakkode

**Pull request merged**